scanpy/tools/_leiden.py

```diff
@@ -134,22 +146,41 @@ def leiden(
             adjacency=adjacency,
         )
     # convert it to igraph
+    if not use_leidenalg and directed:
+        raise ValueError(
+            "Cannot use igraph's leiden implemntation with a directed graph."
+        )
     g = _utils.get_igraph_from_adjacency(adjacency, directed=directed)
     # flip to the default partition type if not overriden by the user
-    if partition_type is None:
+    if partition_type is None and use_leidenalg:
         partition_type = leidenalg.RBConfigurationVertexPartition
+    elif not use_leidenalg and partition_type is not None:
+        raise ValueError("Do not pass in partition_type argument when using igraph.")
     # Prepare find_partition arguments as a dictionary,
     # appending to whatever the user provided. It needs to be this way
     # as this allows for the accounting of a None resolution
     # (in the case of a partition variant that doesn't take it on input)
     if use_weights:
-        partition_kwargs["weights"] = np.array(g.es["weight"]).astype(np.float64)
-    partition_kwargs["n_iterations"] = n_iterations
-    partition_kwargs["seed"] = random_state
+        clustering_args["weights"] = (
+            "weight"
+            if not use_leidenalg
+            else np.array(g.es["weight"]).astype(np.float64)
+        )
+    clustering_args["n_iterations"] = n_iterations
+    if use_leidenalg:
+        clustering_args["seed"] = random_state
+    else:
+        random.seed(random_state)
+        igraph.set_random_number_generator(random)
     if resolution is not None:
-        partition_kwargs["resolution_parameter"] = resolution
+        clustering_args[
+            f"resolution{'_parameter' if use_leidenalg else ''}"
+        ] = resolution
```

Comment:
[The igraph docs also say `resolution_parameter`](https://igraph.org/python/doc/api/igraph.Graph.html#community_leiden)

```suggestion
        clustering_args["resolution_parameter"] = resolution
```